test/cace/test_ari_text.c

```diff
@@ -895,6 +895,73 @@ void test_ari_text_decode_lit_typed_rptset(const char *text, enum ari_prim_type_
     ari_deinit(&ari);
 }
 
+TEST_CASE("ari://test/const/hi", ARI_TYPE_CONST)
+TEST_CASE("ari://test/ctrl/hi", ARI_TYPE_CTRL)
+TEST_CASE("ari://test/LITERAL/hi", ARI_TYPE_LITERAL)
+TEST_CASE("ari://test/NULL/hi", ARI_TYPE_NULL)
+TEST_CASE("ari://test/BOOL/hi", ARI_TYPE_BOOL)
+TEST_CASE("ari://test/BYTE/hi", ARI_TYPE_BYTE)
+TEST_CASE("ari://test/INT/hi", ARI_TYPE_INT)
+TEST_CASE("ari://test/UINT/hi", ARI_TYPE_UINT)
+TEST_CASE("ari://test/VAST/hi", ARI_TYPE_VAST)
+TEST_CASE("ari://test/UVAST/hi", ARI_TYPE_UVAST)
+TEST_CASE("ari://test/REAL32/hi", ARI_TYPE_REAL32)
+TEST_CASE("ari://test/REAL64/hi", ARI_TYPE_REAL64)
+TEST_CASE("ari://test/TEXTSTR/hi", ARI_TYPE_TEXTSTR)
+TEST_CASE("ari://test/BYTESTR/hi", ARI_TYPE_BYTESTR)
+TEST_CASE("ari://test/TP/hi", ARI_TYPE_TP)
+TEST_CASE("ari://test/TD/hi", ARI_TYPE_TD)
+TEST_CASE("ari://test/LABEL/hi", ARI_TYPE_LABEL)
+TEST_CASE("ari://test/CBOR/hi", ARI_TYPE_CBOR)
+TEST_CASE("ari://test/ARITYPE/hi", ARI_TYPE_ARITYPE)
+TEST_CASE("ari://test/AC/hi", ARI_TYPE_AC)
+TEST_CASE("ari://test/AM/hi", ARI_TYPE_AM)
+TEST_CASE("ari://test/TBL/hi", ARI_TYPE_TBL)
+TEST_CASE("ari://test/EXECSET/hi", ARI_TYPE_EXECSET)
+TEST_CASE("ari://test/RPTSET/hi", ARI_TYPE_RPTSET)
+TEST_CASE("ari://test/OBJECT/hi", ARI_TYPE_OBJECT)
```

Comment:
These are valuable test inputs but the correct behavior should actually be to treat these as invalid. While they are all using valid built-in types none of these are valid AMM _object_ types. This was a defect in the original logic, do you mind looking into it? We might need to have more internal flags associated with the built-in type text.

Reply:
No problem, I'll look into it. But I do want to make sure we're on the same page with these changes. So to confirm, for AMM object types we want to accept the following (from https://github.com/JHUAPL-DTNMA/dtnma-tools/blob/c7ef3dddd6db598385e7b39777d409be2e147308/src/cace/ari/type.h#L69):

```
    ARI_TYPE_IDENT   = -1,
    ARI_TYPE_CONST   = -2,
    ARI_TYPE_CTRL    = -3,
    ARI_TYPE_EDD     = -4,
    ARI_TYPE_OPER    = -6,
    ARI_TYPE_SBR     = -8,
    ARI_TYPE_TBR     = -10,
    ARI_TYPE_VAR     = -11,
    ARI_TYPE_TYPEDEF = -12,
```

And we want to reject the other ARI types, including `ARI_TYPE_OBJECT`, correct?

Reply:
Yes exactly, and since these are part of the compiled code a validity check could just be the range of valid (for now) code points with a comment about updating the range if new code points are allocated.